src/main/kotlin/com/stefan_grafberger/streamdq/anomalydetection/strategies/windowfunctions/OnlineNormalAggregate.kt

```diff
@@ -0,0 +1,94 @@
+package com.stefan_grafberger.streamdq.anomalydetection.strategies.windowfunctions
+
+import com.stefan_grafberger.streamdq.anomalydetection.model.dto.NormalStrategyResultDto
+import com.stefan_grafberger.streamdq.checks.AggregateConstraintResult
+import org.apache.flink.api.common.functions.AggregateFunction
+import org.apache.flink.api.java.tuple.Tuple7
+import kotlin.math.sqrt
+
+/**
+ * Online normal aggregate function for detecting anomaly.
+ * detecting anomalies based on a customized bound on the
+ * number of standard deviations they are allowed to be
+ * different from the mean
+ *
+ * This aggregate function is used in
+ * [com.stefan_grafberger.streamdq.anomalydetection.strategies.impl.OnlineNormalStrategy]
+ *
+ * @see <a href="https://fanf2.user.srcf.net/hermes/doc/antiforgery/stats.pdf">Incremental calculation of variance</a>
+ * @since 1.0
+ */
+class OnlineNormalAggregate(
+        private val lowerDeviationFactor: Double? = 3.0,
+        private val upperDeviationFactor: Double? = 3.0,
+) : AggregateFunction<AggregateConstraintResult,
+        Tuple7<Double, Double, Double, Double, Double, Double, Long>,
+        NormalStrategyResultDto> {
+
+    private var currentValue = 0.0
+
+    /**
+     * accumulator used to preserve current mean of aggregate values,
+     * current variance and count of the elements
+     * Tuple7
+     * (lastMean f0, currentMean f1, lastVariance f2,
+     * currentVariance f3, lastSn f4, sn f5, count f6)
+     */
+    override fun createAccumulator(): Tuple7<Double, Double, Double, Double, Double, Double, Long> {
+        return Tuple7(0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0L)
+    }
+
+    override fun add(
+            aggregateConstraintResult: AggregateConstraintResult,
+            acc: Tuple7<Double, Double, Double, Double, Double, Double, Long>
+    )
+            : Tuple7<Double, Double, Double, Double, Double, Double, Long> {
+
+        currentValue = aggregateConstraintResult.aggregate!!
+
+        acc.f0 = acc.f1
```

Comment:
Same comment as in the previous code. Do you think with the suggested changes it would be easier to work with, or do you think this is fine as it is?